### ISSUE TITLE
[Perf] Optimize 1D group convolution

### ISSUE THREAD

**Issue by awni** [ml-explore/mlx#2369]
Here's a benchmark. This op comes up in mamba layers and several other places that use 1D depthwise convolutions. Right now it's almost 2x faster to do the blown up multiply followed by a sum, I believe our current depthwise specialization for metal does not handle the case below.

```python
import time
import mlx.core as mx
import mlx.nn as nn

D = 2048
K = 3

conv_l = nn.Conv1d(
    in_channels=D,
    out_channels=D,
    kernel_size=K,
    groups=D,
    bias=False,
)

w = conv_l.weight
x = mx.random.normal((1, 1, D))
state = mx.zeros((1, K - 1, D))

def conv_ops(x, state):
    x = mx.concatenate([state, x], axis=-2)
    return mx.sum(
        x.transpose(0, 2, 1) * w[:, :, 0],
        axis=-1,
        keepdims=True,
    ).reshape(1, 1, -1)

def conv(x, state):
    x = mx.concatenate([state, x], axis=-2)
    return conv_l(x)

def timeit(fn):
    def looped_fn(x):
        for _ in range(20):
            x = fn(x, state)
        return x

    for _ in range(100):
        mx.eval(looped_fn(x))

    tic = time.time()
    for _ in range(100):
        mx.eval(looped_fn(x))
    toc = time.time()
    print(toc - tic)

timeit(conv)
timeit(conv_ops)
```

**Comment by mhadiniknam** [comment]
Hi there 
I did some experiements on 'mlx/backend/gpu/metal/conv.cpp'
I just make remove that condition of the output matrix dim to be less than 8 in 'dispatch_conv_2D_gpu' second if statement (to use depthwise in dispatch for 1D cases)

and put that check inside the kernel definition on 'mlx/backend/metal/kernels/conv.metal'

That makes the Conv 10x faster but that assumption of running on the 8 by 4 tile make it fail on some test example
I should spend sometime find out what to do with handling that dimension

**Comment by AKLABZ** [comment]
The testing done by @mhadiniknam seem to work as stated, but I assume there is a way to confirm tile compatibility - then provide some sort of trimming/padding methods for cases that don't fit - or skipping threads entirely but that might not be appropriate if people are working with such small output shape.

I'm down to look into more as well but like I said I made the same changes as @mhadiniknam and ran into the same issue. I just wanna know what approach you think would be better for this before I start, or if you think this wouldn't solve the issue then that's fine too.
